rqt_rotors/src/rqt_rotors/hil_plugin.py

```diff
@@ -0,0 +1,157 @@
+        self.arm = rospy.ServiceProxy(self.STR_MAVROS_ARM_SERVICE_NAME, CommandBool)
+        self.send_command_long = rospy.ServiceProxy(self.STR_MAVROS_COMMAND_LONG_SERVICE_NAME, CommandLong)
+        self.set_mode = rospy.ServiceProxy(self.STR_MAVROS_SET_MODE_SERVICE_NAME, SetMode)
+        
+        # Initialize ROS subscribers and publishers
+        self.sys_status_sub = rospy.Subscriber(self.STR_SYS_STATUS_SUB_TOPIC, State, self.sys_status_callback, queue_size=1)
+
+    def on_set_hil_mode_button_pressed(self):
+        new_mode = self.mav_mode | self.MAV_MODE_FLAG_HIL_ENABLED
+        self.hil_enabled = True
+        self.mav_mode = new_mode
+        self.set_mode(new_mode, '')
+        self._widget.text_mode_hil.setText(self.mav_mode_text(self.hil_enabled))
+
+    def on_arm_button_pressed(self):
+        self.arm(True)
+
+    def on_reboot_autopilot_button_pressed(self):
+        self.send_command_long(False, 246, 1, 1, 0, 0, 0, 0, 0, 0)
+    
+    def sys_status_callback(self, msg):
+        if (not self.connected and msg.connected):
+            self._widget.button_set_hil_mode.setEnabled(True)
+            self._widget.button_arm.setEnabled(True)
+            self._widget.button_reboot_autopilot.setEnabled(True)
+            self.connected = True
+            self.last_heartbeat_time = time.time()
+            self._widget.text_mode_safety_armed.setText(self.mav_mode_text(msg.armed))
+            self._widget.text_mode_guided.setText(self.mav_mode_text(msg.guided))
+            return
+
+        if (((time.time() - self.last_heartbeat_time) >= self.TIMEOUT_HIL_HEARTBEAT) and self.hil_enabled):
+            new_mode = self.mav_mode | self.MAV_MODE_FLAG_HIL_ENABLED
+            self.set_mode(new_mode, '')
+
+        if (self.armed != msg.armed):
+            self.armed = msg.armed
+            self.text_mode_safety_armed.setText(self.mav_mode_text(self.armed))
+            self.button_arm.setEnabled(not(self.armed))
+            self.mav_mode = self.mav_mode | self.MAV_MODE_FLAG_SAFETY_ARMED
+
+        if (self.guided != msg.guided):
+            self.guided = msg.guided
+            self.text_mode_guided.setText(self.mav_mode_text(self.guided))
+
+        self.last_heartbeat_time = time.time()
+
+    def clear_mav_mode(self):
+        count = self._widget.mav_mode_layout.rowCount()
+        for i in range(count):
+            self._widget.mav_mode_layout.itemAt(i, QFormLayout.FieldRole).widget().setText(self.STR_UNKNOWN)
+
+    def mav_mode_text(self, mode_enabled):
+        return self.STR_ON if mode_enabled else self.STR_OFF
+
+    def shutdown_plugin(self):
+        if self.sys_status_sub is not None:
+            self.sys_status_sub.unregister()
+
+    def save_settings(self, plugin_settings, instance_settings):
+        # TODO save intrinsic configuration, usually using:
+        # instance_settings.set_value(k, v)
+        pass
+
+    def restore_settings(self, plugin_settings, instance_settings):
```

Comment:
In this case (and the point above), the 'intrinsic' configuration refers to any internal parameters that get set during the operation of the GUI that, once the GUI is closed by the user, might be desirable to reload once the GUI is open again. Commonly, for example, it can be the name of the namespace that gets specified and to which all the subscribers/publishers point to.

The official ROS tutorial on the rqt plugins includes the 'restore_settings' and 'save_settings' functions but does not implement them either (http://wiki.ros.org/rqt/Tutorials/Writing%20a%20Python%20Plugin).

In our case, these are not really relevant, at least in the present form, so I think for now we can definitely remove them.